smartapps/smartthings/lifx-connect.src/lifx-connect.groovy

```diff
@@ -375,6 +375,14 @@ def devicesList(selector = '') {
         } else if (resp.status == 401) {
             log.warn "Access token is not valid"
             state.lifxAccessToken = null
+        } else if (resp.status == 404 && resp.data.contains('Could not find location_id')) {
+            log.warn "Location is not valid"
+            devices = devicesList()
+            devices.each { device ->
+                if (device.location.id != settings.selectedLocationId && getChildDevice(device.id)) {
+                    app.updateSetting("selectedLocationId", device.location.id)
```

Comment:
I think it would be a good idea to also modify the `settings` map here directly, i.e. `settings.selectedLocationId=device.location.id`,  because I don't think `app.updateSetting()` does that, and we'd wind up updating the setting multiple times for a location with many lights.